models/dataset.go

```diff
@@ -182,7 +183,9 @@ func CreateDataset(reader io.Reader) (*Dataset, error) {
 		return nil, errors.New("Failed to parse json body")
 	}
 
-	if dataset.State == "" {
+	// Overwrite state to created if request method is POST
+	if method == PostMethod {
```

Comment:
Consider moving this logic out of this function. Having the method passed into the function feels a bit strange. either have the POST specific logic alongside the function call, or create a wrapper function to add the POST functionality